src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -219,19 +238,47 @@ internal set
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
         public void RunClassInitialize(TestContext testContext)
         {
-            // If no class initialize return
-            if (this.ClassInitializeMethod == null)
+            // If no class initialize and no base class initialize, return
+            if (this.ClassInitializeMethod is null)
             {
-                return;
+                if (this.BaseClassInitializeMethodsQueue.Count == 0)
+                {
+                    return;
+                }
             }
 
             if (testContext == null)
             {
                 throw new NullReferenceException(Resource.TestContextIsNull);
             }
 
-            // If class initialization is not done, then do it.
-            if (!this.IsClassInitializeExecuted)
+            MethodInfo initializeMethod = null;
+            try
+            {
+                // ClassInitialize methods for base classes are called in reverse order of discovery
+                // Base -> Child TestClass
+                var baseClassInitializeStack = new Stack<MethodInfo>(this.BaseClassInitializeMethodsQueue);
+                while (baseClassInitializeStack.Count > 0)
+                {
+                    initializeMethod = baseClassInitializeStack.Pop();
+                    if (initializeMethod.GetCustomAttribute<ClassInitializeAttribute>().InheritanceBehavior
```

Comment:
looks like we already have this check when we add methodInfo's to the queue. We could skip this check altogether.

Reply:
true, already removed!